### ISSUE TITLE
[bug] Window overlap causes image to disappear when it shouldn't

### ISSUE THREAD

**Issue by benlubas** [3rd/image.nvim#49]
Here I have an image rendering with the markdown integration in the buffer.  I have these two options set in my config:

```lua
window_overlap_clear_enabled = true, -- toggles images when windows are overlapped
window_overlap_clear_ft_ignore = { "cmp_menu", "cmp_docs", "" },
```

and the molten output window here does have a filetype, so it _should_ cause the image to disappear if the two overlap.

My issue is that this output window wouldn't overlap the image if it didn't disappear, and in some cases never overlaps the area that was occupied by the image, so I don't think the image should be disappearing here.

https://github.com/3rd/image.nvim/assets/56943754/f4860495-e15e-41d2-aca9-21c4ed5da544

My first guess would be, make sure we're checking against the rendered geometry instead of the full image size. But I'll look into this more tomorrow.

**Comment by 3rd** [comment]
Yep, the current version is very naive for this, been thinking of cropping the images properly for floating panes like Lazy's as well.

**Comment by 3rd** [comment]
Hey, can you make a small repro script for this, if it still happens? :heart:

**Comment by benlubas** [comment]
Open a markdown file with this content:

```
![nvim logo](https://gist.ro/s/remote.png)

:lua local win = vim.api.nvim_open_win(0, false, { relative="win", row = 10, col = 1, width = 50, height = 3, })
```
step 0: make sure that this image doesn't take up more than 9 rows when rendered. If it does, increase the value of `row`.
1. open the file, wait for the image to show up.
2. visually select the entire `:lua ...` line, and type `:so<CR>` to run that line, which creates a floating window below the image.
